apps/anomalist/gp_detector.py

```diff
@@ -13,52 +13,54 @@
 
 from etl import grapher_model as gm
 
-log = structlog.get_logger()
-
-
-class AnomalyDetector:
-    anomaly_type: str
+from .detectors import AnomalyDetector
 
+log = structlog.get_logger()
 
-class SampleAnomalyDetector(AnomalyDetector):
-    anomaly_type = "sample"
 
-    def get_score_df(self, df: pd.DataFrame, variables: list[gm.Variable]) -> pd.DataFrame:
-        score_df = cast(pd.DataFrame, df.sample(1))
-        score_df.iloc[0, :] = [random.normalvariate(0, 1) for _ in range(df.shape[1])]
-        return score_df
+class AnomalyGaussianProcessOutlier(AnomalyDetector):
+    anomaly_type = "gp_outlier"
 
+    def get_score_df(self, df: pd.DataFrame, variable_ids: List[int], variable_mapping: Dict[int, int]) -> pd.DataFrame:
+        # Create a dataframe of zeros.
```

Comment:
The comment needs to be adjusted. But also, you could indeed just use a dataframe of zeros, and then define the z-score in absolute value. Or is there any benefit in keeping the sign of the deviations?

Reply:
Z-score is converted to absolute value at the very end. You're right that it doesn't matter whether we use zeros or nans. I found nans more useful for visualization purposes when prototyping, so I'm going to keep them, but feel free to change it if you want.